pkg/simple/client/monitoring/metricsserver/metricsserver.go

```diff
@@ -0,0 +1,438 @@
+	var parsedResp monitoring.Metric
+
+	return parsedResp
+}
+
+func (m metricsServer) GetMetricOverTime(expr string, start, end time.Time, step time.Duration) monitoring.Metric {
+	var parsedResp monitoring.Metric
+
+	return parsedResp
+}
+
+var edgeNodeMetrics = []string{"node_cpu_usage", "node_cpu_total", "node_cpu_utilisation", "node_memory_usage_wo_cache", "node_memory_total", "node_memory_utilisation"}
+
+func (m metricsServer) parseErrorResp(metrics []string, err error) []monitoring.Metric {
+	var res []monitoring.Metric
+
+	for _, metric := range metrics {
+		parsedResp := monitoring.Metric{MetricName: metric}
+		parsedResp.Error = err.Error()
+	}
+
+	return res
+}
+
+func (m metricsServer) GetNamedMetrics(metrics []string, ts time.Time, o monitoring.QueryOption) []monitoring.Metric {
+	var res []monitoring.Metric
+
+	opts := monitoring.NewQueryOptions()
+	o.Apply(opts)
+	if opts.Level == monitoring.LevelNode {
+		if !m.metricsAPIAvailable {
+			klog.Warningf("Metrics API not available.")
+			return m.parseErrorResp(metrics, errors.New("Metrics API not available."))
+		}
+
+		edgeNodes, err := m.listEdgeNodes()
+		if err != nil {
+			klog.Errorf("List edge nodes error %v\n", err)
+			return m.parseErrorResp(metrics, err)
+		}
+
+		edgeNodeNamesFiltered := m.filterEdgeNodeNames(edgeNodes, opts)
+		if len(edgeNodeNamesFiltered) == 0 {
+			klog.V(4).Infof("No edge node metrics is requested")
+			return res
+		}
+
+		metricsResult, err := m.getNodeMetricsFromMetricsAPI()
+		if err != nil {
+			klog.Errorf("Get edge node metrics error %v\n", err)
+			return m.parseErrorResp(metrics, err)
+		}
+
+		metricsMap := make(map[string]bool)
+		for _, m := range metrics {
+			metricsMap[m] = true
+		}
+
+		status := make(map[string]v1.NodeStatus)
+		for n, _ := range edgeNodeNamesFiltered {
+			status[n] = edgeNodes[n].Status
+		}
+
+		nodeMetrics := make(map[string]*monitoring.MetricData)
+		for _, enm := range edgeNodeMetrics {
+			_, ok := metricsMap[enm]
+			if ok {
+				nodeMetrics[enm] = &monitoring.MetricData{MetricType: monitoring.MetricTypeVector}
+			}
+		}
+
+		var usage v1.ResourceList
+		var cap v1.ResourceList
+		for _, m := range metricsResult.Items {
+			_, ok := edgeNodeNamesFiltered[m.Name]
+			if !ok {
+				continue
+			}
+
+			m.Usage.DeepCopyInto(&usage)
+			status[m.Name].Capacity.DeepCopyInto(&cap)
+
+			metricValues := make(map[string]*monitoring.MetricValue)
+
+			for _, enm := range edgeNodeMetrics {
+				metricValues[enm] = &monitoring.MetricValue{
+					Metadata: make(map[string]string),
+				}
+				metricValues[enm].Metadata["node"] = m.Name
+				metricValues[enm].Metadata["role"] = "edge"
+			}
+			for _, addr := range status[m.Name].Addresses {
+				if addr.Type == v1.NodeInternalIP {
+					for _, enm := range edgeNodeMetrics {
+						metricValues[enm].Metadata["host_ip"] = addr.Address
+					}
+					break
+				}
+			}
+
+			_, ok = metricsMap["node_cpu_usage"]
```

Comment:
better to use a constant for metrics name `node_cpu_usage` in case of changing in the future

Reply:
Fixed